Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -527,6 +528,26 @@ class BrowserViewController: UIViewController {
             make.top.left.right.equalTo(self.view)
             make.height.equalTo(self.view.safeAreaInsets.top)
         }
+
+        adjustURLBarHeightBasedOnLocationViewHeight()
+    }
+
+    fileprivate func adjustURLBarHeightBasedOnLocationViewHeight() {
+        // Make sure that we have a height to actually base our calculations on
+        guard urlBar.locationContainer.bounds.height != 0 else { return }
+        let locationViewHeight = urlBar.locationView.bounds.height
+        let heightWithPadding = locationViewHeight + 10
+
+        // We have to deactivate the original constraint, and remake the constraint
+        // or else funky conflicts happen
+        urlBarHeight.deactivate()
+        urlBar.snp.makeConstraints { make in
```

Comment:
Wondering if we can use `urlBar.snp.remakeConstraints` instead 🤔

Reply:
That's the first thing I tried and it was doing weird things. I can retry if you think it'll be cleaner and see if I can make it work.

Reply:
**Yeah I agree, lets not go with remake constraint route.**

The reason it breaks is since you are remaking constraints you'll need to add all other constraints otherwise it'll break the view. 

@electricRGB On a sidenote what do you think of this cleanup?

```
        urlBar.snp.makeConstraints { make in
            let height =  heightWithPadding > UIConstants.TopToolbarHeightMax ? UIConstants.TopToolbarHeight : heightWithPadding
            self.urlBarHeight = make.height.equalTo(height).constraint
        }
```

Reply:
yeah, that works. I'll make the changes (and the rename) and push.

Reply:
Thanks @electricRGB 🙂